server/test/services/openapi/v2/Swagger2SchemaGeneratorTest.java

```diff
@@ -0,0 +1,363 @@
+package services.openapi.v2;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import auth.ProgramAcls;
+import com.google.common.collect.ImmutableList;
+import controllers.dev.seeding.SampleQuestionDefinitions;
+import java.util.Optional;
+import java.util.stream.Stream;
+import models.DisplayMode;
+import org.junit.Test;
+import services.LocalizedStrings;
+import services.openapi.OpenApiSchemaSettings;
+import services.openapi.v2.serializers.OpenApiSerializationAsserter;
+import services.program.BlockDefinition;
+import services.program.ProgramDefinition;
+import services.program.ProgramQuestionDefinition;
+import services.program.ProgramType;
+import services.program.StatusDefinitions;
+import services.question.types.QuestionDefinition;
+
+public class Swagger2SchemaGeneratorTest extends OpenApiSerializationAsserter {
+  private static final Stream<QuestionDefinition> ALL_SAMPLE_QUESTION_DEFINITIONS_WITH_IDS_STREAM =
```

Comment:
server/test/support/TestQuestionBank.java might be helpful here, as well as the FakeProgramBuilder utility in server/test/services/export/AbstractExporterTest.java